instrumentation/opentelemetry-sdk-extension-autoconfigure-1.28.0/src/main/java/io/opentelemetry/sdk/autoconfigure/AutoConfiguredOpenTelemetrySdk.java

```diff
@@ -0,0 +1,49 @@
+package io.opentelemetry.sdk.autoconfigure;
+
+import com.newrelic.agent.bridge.AgentBridge;
+import com.newrelic.api.agent.Agent;
+import com.newrelic.api.agent.NewRelic;
+import com.newrelic.api.agent.weaver.MatchType;
+import com.newrelic.api.agent.weaver.Weave;
+import com.newrelic.api.agent.weaver.Weaver;
+import io.opentelemetry.sdk.autoconfigure.spi.internal.DefaultConfigProperties;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Weave(type = MatchType.ExactClass)
+public class AutoConfiguredOpenTelemetrySdk {
+
+    public static AutoConfiguredOpenTelemetrySdkBuilder builder() {
+        final AutoConfiguredOpenTelemetrySdkBuilder builder = Weaver.callOriginal();
+        if (System.getProperty("otel.exporter.otlp.endpoint") == null &&
```

Comment:
This is a decent way to determine if they are trying to configure the SDK themselves, but not perfect. There are signal-specific overrides for the endpoint that they may be setting of the form `otel.exporter.otlp.metrics.endpoint`. If the user sets this and doesn't set `otel.exporter.otlp.endpoint`, then this if-block will be executed, but metric traffic will go to the user-configured endpoint. Not the end of the world, but potentially confusing.

I wonder if we can think of a more reliable signal to trigger this configuration? 🤔